src/org/traccar/notification/EventForwarder.java

```diff
@@ -27,42 +30,60 @@
 import java.util.HashMap;
 import java.util.Map;
+import com.ning.http.client.FluentCaseInsensitiveStringsMap;
 
-public final class EventForwarder {
+public abstract class EventForwarder {
 
-    private String url;
-    private String header;
+    private final String url;
+    private final String header;
+    private final String payloadParamName;
+    private final String additionalParams;
 
     public EventForwarder() {
         url = Context.getConfig().getString("event.forward.url", "http://localhost/");
         header = Context.getConfig().getString("event.forward.header", "");
+        payloadParamName = Context.getConfig().getString("event.forward.paramMode.payloadParamName", "payload");
+        additionalParams = Context.getConfig().getString("event.forward.paramMode.additionalParams", "");
     }
 
     private static final String KEY_POSITION = "position";
     private static final String KEY_EVENT = "event";
     private static final String KEY_GEOFENCE = "geofence";
     private static final String KEY_DEVICE = "device";
 
-    public void forwardEvent(Event event, Position position) {
+    public final void forwardEvent(Event event, Position position) {
 
         BoundRequestBuilder requestBuilder = Context.getAsyncHttpClient().preparePost(url);
+        requestBuilder.setBodyEncoding(StandardCharsets.UTF_8.name());
+
+        requestBuilder.addHeader("Content-Type", getContentType());
 
-        requestBuilder.addHeader("Content-Type", "application/json; charset=utf-8");
         if (!header.equals("")) {
-            String[] headerLines = header.split("\\r?\\n");
-            for (String headerLine: headerLines) {
-                String[] splitedLine = headerLine.split(":", 2);
-                if (splitedLine.length == 2) {
-                    requestBuilder.setHeader(splitedLine[0].trim(), splitedLine[1].trim());
-                }
-            }
+            FluentCaseInsensitiveStringsMap params = new FluentCaseInsensitiveStringsMap();
+            params.putAll(splitParams(header, ":"));
+            requestBuilder.setHeaders(params);
         }
 
-        requestBuilder.setBody(preparePayload(event, position));
+        setContent(event, position, requestBuilder);
         requestBuilder.execute();
     }
 
-    private byte[] preparePayload(Event event, Position position) {
+    protected Map<String, List<String>> splitParams(String params, String separator) {
+
+        String[] splitedLine;
+        Map<String, List<String>> paramsMap = new HashMap<>();
+        String[] paramsLines = params.split("\\r?\\n");
+
+        for (String paramLine: paramsLines) {
+            splitedLine = paramLine.split(separator, 2);
+            if (splitedLine.length == 2) {
+                paramsMap.put(splitedLine[0].trim(), Arrays.asList(splitedLine[1].trim()));
+            }
+        }
+        return paramsMap;
+    }
+
+    protected String prepareJsonPayload(Event event, Position position) {
```

Comment:
Why is this in the base class as well?

Reply:
it converts event data to json data, so it's used to create all the body content for "JsonMode" and the payload parameter for "paramMode"

Reply:
You reminded me my problem with this feature. Can we just have a pull request for improvements to formatter and context classes? I don't really like sending JSON as encoded form parameter. I don't see any good use case for it.

Reply:
Even spending considerable efforts on discussions and explanations we still have trivial misunderstandings, i changed the course of this implementation about 3 times. 

Anyway, i see your point, i made some changes, please see the main thread.